src/gSP.cpp

```diff
@@ -1685,9 +1675,11 @@ void gSPModifyVertex( u32 _vtx, u32 _where, u32 _val )
 				}
 
 				vtx0.y = -(vtx0.y - gSP.viewport.vtrans[1]) / gSP.viewport.vscale[1];
-				if (gSP.viewport.vscale[1] < 0)
-					vtx0.y = -vtx0.y;
+				//if (gSP.viewport.vscale[1] < 0)
+				//	vtx0.y = -vtx0.y;
 				vtx0.y *= vtx0.w;
+			*/
+			if (false) {
 			} else {
```

Comment:
Please cleanup this part: remove that hack if it is not necessary anymore (remove all the related code) and remove this "if".

Reply:
I deliberately left this in place because I didn't know what the hack was for. I know the coordinate inversions in the commented part (mentioned in your annotation just above) were giving me trouble, but I can't remember if the rest of the hack was problematic.

Do you know what the hack is for? Should I reenable it or discard it? I'll double check later, but if I remember correctly I saw no issues both with the hack and without it.

Reply:
The hack was enabled for three games, see RSP_Init().
However, you disabled the main branch, which is without hack.
That is now it works as if the hack is always enabled.
May be it is correct solution now, because previously I applied reversed viewport transformation to the provided screen space coordinate. All games, which use gSPModifyVertex command should be checked. Unfortunately, I don't remember  these games. @olivieryuyu, could you help us with it? You have good memory :)

Reply:
gSPModifyVertex: Tarzan menu options or Jeremy McGrath Supercross 2000 in menu i believe. I think so All Star baseball games as well.

Reply:
> However, you disabled the main branch, which is without hack.

Mmm, I had not realized this, I must have done it unintentionally. I'll take a closer look at the situation.

Reply:
The situation seems a bit complicated. The reverse viewport transformation is no longer needed, so I removed it. 
I am not sure about the `dwnd().isAdjustScreen()` part (seems it was introduced here https://github.com/gonetz/GLideN64/commit/038c74af123f068a36e7e6f29d7fbe28ee79618a).

Tarzan works fine without the hack, but forcing `w` to `1.0f` creates issues because `z` needs to be divided by it. The hack is still needed in Rat Attack for the projector effect in the intro and in Resident Evil 2 for the rotating Nintendo logo. It seems `w` is zero in this games, which creates issues when we multiply by it in the vertex shader and OpenGL performs division by `w`. Thus neither forcing to `1.0f` nor not forcing it seems the correct approach. I will leave the hack there for now, I don't understand the situation well enough to find a good solution.

Reply:
Not sure it helps but:

gSPModifyVertex uses the data of a precalculated vertex and includes new screenspace coordinates in it (usually), new RGBA or new S&T.

The W of the vertex remains the very same than the first vertex calculated.

So your code you keep the W of the previous vertex pointed out by the commmand.

Reply:
@olivieryuyu The problem I have is that I don't know at what stage the coordinates are injected. See [OpenGL's vertex post-processing pipeline](https://www.khronos.org/opengl/wiki/Vertex_Post-Processing).

Firstly, primitives must be assembled in order to perform clipping, as one vertex may clip into different position based on its neighbours. This makes it specially difficult to refer to vertices with indices, like gSPModifyVertex does, after clipping has occurred, because new vertices are generated dinamically.

So my GUESS is that the data is modified prior to all this, when only the vertices programmed by the developer exist. However, gSPModifyVertex uses screen space coordinates, so they need no perspective divide nor viewport transform. Assuming the data is injected prior to the vertex post-processing, the RDP should implicitly disable perspective divide and viewport transform. 

The latter should not be too hard to emulate (as it is done in vertex shader now) but the former is complicated. OpenGL enforces perspective divide, and changing w to 1 can be problematic if XY don't need the perspective divide but Z does (which seems to happen in Tarzan). Multiplying XY by w works in this case, as the perspective divide will then restore them to original values, but only because w is not 0. To make it worse, other games (like Rat Attack or Resident Evil 2) use gSPModifyVertex when w=0, so multiplying XY by w doesn't work for those games.

It might be possible to emulate this without hacks if the whole RSP is emulated in software, i.e., if the perspective divide and viewport tranform are also done in software. Then it would be possible to pass `(x_screen, y_screen, z_screen, 1)` to OpenGL, and gSPModifyVertex would only have to replace this x_screen and y_screen values. Anyway, I feel all of this is a bit out of the scope of the pull request and it is better to keep the hack for now, maybe consider trying to implement the idea later.

Reply:
gSPModifyVertex() was adapted.